lib/utils.pm

```diff
@@ -190,6 +191,16 @@ sub is_casp {
     return check_var('FLAVOR', $flavor);
 }
 
+# Checks if hypervisor is xen with optional type [pv|hvm] parameter
+sub is_xen {
+    my $type = shift;
+    return 0 unless check_var('VIRSH_VMM_FAMILY', 'xen');
+    return 1 unless $type;
+
+    return check_var('VIRSH_VMM_TYPE', 'linux') if $type eq 'pv';
+    return check_var('VIRSH_VMM_TYPE', $type);
+}
```

Comment:
XEN offers two virtualization types Paravirtual (PV) and Hardware Virtual Machine (HVM).
PV is called "linux" because that's the name they used in xml definition of the VM.
I think PV is more intuitive - but "linux" can be also used.